tests/test_config.py

```diff
@@ -1,32 +1,50 @@
 import os
 
+potential_paths = [
+    'tests/',
+    '../tests/'
+    'tap-square/tests/',
+    '../tap-square/tests/',
+]
+
+def go_to_tests_directory():
+    for path in potential_paths:
+        if os.path.exists(path):
+            os.chdir(path)
+            return os.getcwd()
+    raise NotImplementedError("This check cannot run from {}".format(os.getcwd()))
+
+##########################################################################
+### TEST
+##########################################################################
 
-cwd = os.getcwd()
-if cwd not in {'root/project', '/opt/code/tap-square'}:
-    print("WARN: This script is meant to run from the top level directory of the tap.")
+print("Acquiring path to tests directory.")
+cwd = go_to_tests_directory()
 
 print("Reading in filenames from tests directory.")
-print("Parsing directory for tests.")
-files = [
-    name for name in os.listdir(cwd + '/tests')
-    if 'test' == name[:4] and '.py' == name[-3:] and \
-    name not in ['test_client.py', 'test_config.py']
-]
-print("\tFiles found: {}".format(files))
+files_in_dir = [name for name in os.listdir(cwd)]
+
+print("Dropping files that are not of the form 'test_<feature>.py'.")
+test_files_in_dir = [fn for fn in files_in_dir if 'test' == fn[:4] and '.py' == fn[-3:]]
 
-print("Reading contents of circle config")
-with open(cwd + "/.circleci/config.yml", "r") as config:
+print("Dropping test_client.py from test files.")
+test_files_in_dir.remove('test_client.py')
+
+print("Files found: {}".format(test_files_in_dir))
+
+print("Reading contents of circle config.")
+with open(cwd + "/../.circleci/config.yml", "r") as config:
     contents = config.read()
 
-# Check that each file in the directory can be found in a run black in the circle config
 print("Parsing circle config for run blocks.")
-runs = contents.replace(' ', '').replace('\n', '').split('-run:') # separate into run blocks
-matches = {f: False for f in files}
-for m in matches.keys():
-    print("Verifying {} is running in circle.".format(m))
-    if any([m in run for run in runs]):
-        matches[m] = True
-
-# Verify all files were found
-assert all(matches.values()), "The following tests are not running in circle:\t{}".format([k for k, v in matches.items() if not v])
+runs = contents.replace(' ', '').replace('\n', '').split('-run:')
+
+print("Verify all test files are executed in circle...")
+file_found = {f: False for f in test_files_in_dir}
+for filename in file_found.keys():
```

Comment:
You don't need to do .keys(). If you just do `for filename in file_found` it works, iterating over keys is the default iterator for a dict.